NPE while validating VMware creds

### What steps did you take and what happened?

got below error while validating vmwarecreds

```
{"controller": "openstackcreds", "controllerGroup": "[vjailbreak.k8s.pf9.io](http://vjailbreak.k8s.pf9.io/)", "controllerKind": "OpenstackCreds", "OpenstackCreds": {"name":"a2c6cfe8-4888-47de-b858-ac7bc9f2fbab","namespace":"migration-system"}, "namespace": "migration-system", "name": "a2c6cfe8-4888-47de-b858-ac7bc9f2fbab", "reconcileID": "b653826d-ace3-43e9-b35a-3dacf61cb4ab"}
2025-02-20T11:28:21Z	INFO	Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference	{"controller": "migrationtemplate", "controllerGroup": "[vjailbreak.k8s.pf9.io](http://vjailbreak.k8s.pf9.io/)", "controllerKind": "MigrationTemplate", "MigrationTemplate": {"name":"76ad5bc5-a6bd-426f-964c-d8ae3af48af1","namespace":"migration-system"}, "namespace": "migration-system", "name": "76ad5bc5-a6bd-426f-964c-d8ae3af48af1", "reconcileID": "3d8fec4b-7ab7-4bdf-9d52-7658e5cbeb42"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x2d8 pc=0x19433b7]
goroutine 123 [running]:
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1())
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.2/pkg/internal/controller/controller.go:111 +0x1e5
panic({0x1bdcd80?, 0x332f820?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
[github.com/platform9/vjailbreak/k8s/migration/internal/controller.GetAllVMs({0x22b7ee8](http://github.com/platform9/vjailbreak/k8s/migration/internal/controller.GetAllVMs(%7B0x22b7ee8), 0xc0006a4510}, 0x0?, {0xc00027e160, 0x8})
	/workspace/internal/controller/vmwarecreds_controller.go:245 +0x557
[github.com/platform9/vjailbreak/k8s/migration/internal/controller.(*MigrationTemplateReconciler).Reconcile(0xc0004a70e0](http://github.com/platform9/vjailbreak/k8s/migration/internal/controller.(*MigrationTemplateReconciler).Reconcile(0xc0004a70e0), {0x22b7ee8, 0xc0006a4510}, {{{0xc00027e140?, 0x5?}, {0xc00014ef90?, 0xc00073bd50?}}})
	/workspace/internal/controller/migrationtemplate_controller.go:85 +0x35b
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x22bc070](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x22bc070)?, {0x22b7ee8?, 0xc0006a4510?}, {{{0xc00027e140?, 0xb?}, {0xc00014ef90?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.2/pkg/internal/controller/controller.go:114 +0xb7
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001d33f0](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001d33f0), {0x22b7f20, 0xc000450960}, {0x1d61b40, 0xc000592180})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.2/pkg/internal/controller/controller.go:311 +0x3bc
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001d33f0](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001d33f0), {0x22b7f20, 0xc000450960})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.2/pkg/internal/controller/controller.go:261 +0x1c9
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2())
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.2/pkg/internal/controller/controller.go:222 +0x79
created by [sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2) in goroutine 71
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.2/pkg/internal/controller/controller.go:218 +0x486
```

### What did you expect to happen?

vmware creds should get validated successfully

### Environment

v0.1.2

### vCenter version

any

### Anything else you would like to add?

_No response_

--- a/v2v-helper/migrate/migrate.go
+++ b/v2v-helper/migrate/migrate.go
@@ -590,6 +590,10 @@ func (migobj *Migrate) CreateTargetInstance(vminfo vm.VMInfo, flavorId string) e
 				return fmt.Errorf("failed to get network: %s", err)
 			}
 
+			if network == nil {
+				return fmt.Errorf("network not found")
+			}
+
 			ip := ""
 			if len(vminfo.Mac) != len(vminfo.IPs) {
 				ip = ""

--- a/v2v-helper/pkg/utils/openstackopsutils.go
+++ b/v2v-helper/pkg/utils/openstackopsutils.go
@@ -358,6 +358,11 @@ func (osclient *OpenStackClients) CreatePort(network *networks.Network, mac, ip,
 	}
 	log.Printf("Port with MAC address %s does not exist, creating new port\n", mac)
 	log.Println("Trying with same IP address: ", ip)
+
+	// Check if subnet is valid to avoid panic.
+	if len(network.Subnets) == 0 {
+		return nil, fmt.Errorf("no subnets found for network: %s", network.ID)
+	}
 	port, err := ports.Create(osclient.NetworkingClient, ports.CreateOpts{
 		Name:       "port-" + vmname,
 		NetworkID:  network.ID,

